src/Streamly/Internal/Data/Stream/IsStream/Expand.hs

```diff
@@ -460,29 +460,29 @@ mergeAsyncByM f m1 m2 =
 -- therefore provide many times better performance.
 --
 -- @since 0.7.0
-{-# INLINE concatUnfold #-}
-concatUnfold ::(IsStream t, Monad m) => Unfold m a b -> t m a -> t m b
-concatUnfold u m = fromStreamD $ D.concatUnfold u (toStreamD m)
+{-# INLINE unfoldMany #-}
+unfoldMany ::(IsStream t, Monad m) => Unfold m a b -> t m a -> t m b
+unfoldMany u m = fromStreamD $ D.unfoldMany u (toStreamD m)
```

Comment:
We need to retain the old function as it is, just add a `DEPRECATED` pragma. Introduce a new `unfoldMany` function and move the doc from `concatUnfold` to the new function, and change the since notation to 0.8.0. In `concatUnfold` just say "Same as unfoldMany".